fr.esrf.tango.generator.xtend/src/fr/esrf/tango/pogo/generator/python/PythonUtils.xtend

```diff
@@ -274,6 +274,47 @@ class PythonUtils {
     	return labelList
     }
     
+    def enumLabelWithInvalidChars (Attribute attr){
+    	var enumVal = 0
+    	var String labelList = ""
+    	for(label: attr.enumLabels){
+    		labelList = labelList + "(\""+ label + "\", " + enumVal.toString +"),\n"
+    		enumVal = enumVal +1
+   		}
+	    return labelList
+    }
+    
+    def String checkEnumLabels (Attribute attr){
+    	var flag = "valid"
+    	for(label: attr.enumLabels){
+    	 	if(label.contains("-")||label.contains("!")||label.contains("#")||label.contains("@")||label.contains("%")||label.contains("$")){
```

Comment:
Python identifiers cannot start with a digit either.  Not sure if Tango allows labels like `"345_is_a_bad_label"`.

Reply:
Agree. Tango does allow labels starting with digits. I have considered digit at start of the label as invalid label.